rest/server.go

```diff
@@ -515,46 +517,73 @@ func (server *HTTPServer) StopHTTP() {
 	)
 }
 
-// NewMux creates a new http handler for the http server, typically you only
-// need to call this directly for tests.
-func (server *HTTPServer) NewMux() http.Handler {
-	var router Router
-	if server.cors != nil {
-		router = NewRouterWithCORS(notFoundHandler, server.cors)
-	} else {
-		router = NewRouter(notFoundHandler)
+func corsHandler(handler http.Handler, corsOptions *CORSOptions) http.Handler {
+	if corsOptions == nil {
+		return handler
 	}
 
-	for _, f := range server.services {
-		f.Register(router)
-	}
-	logger.Debugf("api=NewMux, service=%s, service_count=%d",
-		server.Name(), len(server.services))
+	c := cors.New(cors.Options{
+		AllowedOrigins:         corsOptions.AllowedOrigins,
+		AllowOriginFunc:        corsOptions.AllowOriginFunc,
+		AllowOriginRequestFunc: corsOptions.AllowOriginRequestFunc,
+		AllowedMethods:         corsOptions.AllowedMethods,
+		AllowedHeaders:         corsOptions.AllowedHeaders,
+		ExposedHeaders:         corsOptions.ExposedHeaders,
+		MaxAge:                 corsOptions.MaxAge,
+		AllowCredentials:       corsOptions.AllowCredentials,
+		OptionsPassthrough:     corsOptions.OptionsPassthrough,
+		Debug:                  corsOptions.Debug,
+	})
+	return c.Handler(handler)
+}
 
-	var err error
-	httpHandler := router.Handler()
+func authzHandler(handler http.Handler, authz Authz) (http.Handler, error) {
```

Comment:
I'm not a fun to intro bunch of small functions that are used only once in the code.

Reply:
i feel it simplifies the code in the caller. otherwise the caller would have to do something like:
```
var authzHandler http.Handler
if server.authz != nil {
   //code to set authzHandler
} else {
   authzHandler = corsHandler
}
// ....rest....
``` 

while i agree it is not a "big" savings; i think it makes using the authz handler the same as cors handler, and other handlers.... if you feel strongly, I can remove it.